tools/richlist.js

```diff
@@ -423,27 +480,36 @@ function prepareJsonAddress(json, defaultType = 0) {
   return accounts;
 }
 
-function readJsonAccounts(json, blockNumber, callback, defaultType = 0) {
+async function readJsonAccounts(json, blockNumber, callback, defaultType = 0) {
   const data = prepareJsonAddress(json, defaultType);
   const accounts = Object.keys(data);
   console.log(`* update ${accounts.length} genesis accounts...`);
-  async.eachSeries(accounts, (account, eachCallback) => {
-    web3.eth.getBalance(account, (err, balance) => {
-      if (err) {
-        console.log(`ERROR: fail to getBalance(${account})`);
-        return eachCallback(err);
-      }
 
-      data[account].balance = parseFloat(web3.utils.fromWei(balance, 'ether'));
-      eachCallback();
+  // batch rpc job
+  let batch = new web3.BatchRequest();
+  // normally, the request size of batch getBalance() of all accounts is not bigger than 128kB.
+  // simply getBalance at once using rpc batch job.
+  for (let i = 0; i < accounts.length; i++) {
+    let account = accounts[i];
+    batch.add(web3.eth.getBalance.request(account));
+  }
+
+  try {
+    // web3 2.0.0 case
```

Comment:
By using web3 2.0, you should remove web3explorer(web3);

Reply:
Same with this PR https://github.com/ethereumclassic/explorer/pull/328, you should also replace web3-explorer package which is used for parity specific RPC calls.

Reply:
Ref: https://web3js.readthedocs.io/en/v1.2.1/web3.html#extend